internal/policy/verify.go

```diff
@@ -682,6 +682,14 @@ func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, en
 		return err
 	}
 
+	if entry.TargetID != tagObj.Target && entry.TargetID != tagObj.Hash {
+		return fmt.Errorf("verifying RSL entry failed, %s", ErrUnauthorizedSignature.Error())
+	}
+
+	if entry.RefName != "refs/tags/"+tagObj.Name {
+		return fmt.Errorf("verifying RSL entry failed, tag name has been changed")
```

Comment:
it may have changed or it may not have, it may actually be the case that refs/tags/<x> points to tag <y>. Changed is more indicative that the same tag name has a different target. I'd rephrase this to just say they don't match.

FWIW, I think Git may have some default protections for this submitted as part of the original omitting commits work.